syntax = "proto3";

option
    java_package = "com.huasheng.sa.quant.open.sdk.protobuf.hq.notify";
option java_outer_classname = "StatisticsNotifyProto";
message StatisticsNotify {
  string stockCode = 1;
  int64 tradingDay = 2;//202211280905
  string lastPrice = 3;//最新价：145.38
  string volume = 4;//成交数量（不包含单位 ）
  string turnOver = 5;//成交额
  string quantityRelativeRatio = 6;
  string preClose = 7;//昨收价
  string high = 8;//最高价
  string low = 9;//最低价
  string open = 10;//开盘价
  string midPrice = 11;//中间价
  int64 timestamp = 12; // 时间戳（tradingDay） 
  string fiveMinPriceLimit = 13;//五分钟涨跌幅
  string turnOverRatio = 14;//换手率
  int32 dataType = 15;//证券类型
  string totalMarketCap = 16;//总市值

  float pe = 17;//市盈率
  float pb = 18;//市净率
  int32 mktTmType = 19; //1:盘中 -1:盘前 -2:盘后 包含负数 int32 序列化直接数比较大 -1 占用了11个字节
  string status = 20;//股票状态码
  int64 sourceTime = 21;//供应商时间戳
  int64 sourceRecvTime = 22;//接收端收到时间戳
  int64 sourcePushTime = 23;//推送端推送时间戳
  int32 filedUpdateFlag = 24;//表示交易所消息是否有影响快照字段的变更，用二进制位来标记多个字段最新价：1，买档价格：1<<1，买档量：1<<2 卖档价格：1<<3，卖档量：1<<4
  string tradeTime = 25; //yyyyMMddHHmmss
}
